prog/vnet/update_load_balancer_node.rb

```diff
@@ -129,22 +133,30 @@ def generate_lb_ip_set_definition(neighbor_vms)
       "elements = {#{neighbor_vms.map(&:private_ipv6).join(", ")}}"]
   end
 
-  def generate_lb_map_defs_ipv4(current_port)
-    load_balancer.active_vm_ports
+  def generate_lb_map_defs(current_port)
+    items = load_balancer.active_vm_ports
       .select { |vm_port| vm_port.load_balancer_port.dst_port == current_port.dst_port }
-      .map
-      .with_index { |vm_port, index| "#{index} : #{vm_port.load_balancer_vm.vm.private_ipv4} . #{(vm_port.vm.id == vm.id) ? vm_port.load_balancer_port.dst_port : vm_port.load_balancer_port.src_port}" }
+      .map do |vm_port|
+        address = yield vm_port
+        port = (vm_port.load_balancer_vm.vm_id == vm.id) ? vm_port.load_balancer_port.dst_port : vm_port.load_balancer_port.src_port
+        [address.to_s, port.to_i]
+      end
+
+    items.sort_by { |address, port| [address, port] }.map.with_index do |(address, port), index|
```

Comment:
It doesn't look like `sort_by` is needed here.  Since the array was generated by  the `map` above, and it is not shared, you can use `sort!` and `map!`:

```suggestion
    items.sort!.map!.with_index do |(address, port), index|
```